### PR TITLE
SWARM-787 - Allow specification of a main() from a @DefaultDeployment.

### PR DESCRIPTION
**Pull request by bobmcwhirter**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

---
## Motivation

To cleanly handle testing better, and avoid duplication, allow
provisioning of a main() through an attribute on the @DefaultDeployment
annotation.
## Modifications

Adding @DefaultDeployment(main=MyMain.class) capability.
## Result

Instead of requiring a @CreateSwarm which may replicate the logic
performed within an exist main(), just allow pointing to the existing
main().

### CODE REVIEW

**Comment by gastaldi** [comment]
Retest this please

**Review by gastaldi — APPROVED** [review]
Looks good to me

File: arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/DefaultDeployment.java [hunk 1]
```diff
@@ -37,4 +37,5 @@ public String toString() {
 
     Type type() default Type.JAR;
     boolean testable() default true;
+    Class<?> main() default Object.class;
```
Comment (gastaldi) [inline arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/DefaultDeployment.java]: Where do you read this information?

Comment (gastaldi) [reply arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/DefaultDeployment.java]: Ah nevermind, just saw in DefaultDeploymentScenarioGenerator

Comment (bobmcwhirter) [reply arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/DefaultDeployment.java]: Which passes it along to UberJarSimpleContainer.

**Review by kenfinnigan — APPROVED** [review]
CI passed, still want to hold?

**Comment by bobmcwhirter** [comment]
Merge away.

**Pull request merged**